src/libUtils/RootComputation.cpp

```diff
@@ -87,13 +91,21 @@ TxnHash ComputeRoot(const list<Transaction>& receivedTransactions,
                     const list<Transaction>& submittedTransactions) {
   LOG_MARKER();
 
+  if (receivedTransactions.empty() && submittedTransactions.empty()) {
```

Comment:
I think we can just make `ConcatTranAndHash` do the check for all the inputs if they are empty?  That way the risk is lower of us failing to check if one of the containers is empty.